test/edu/stanford/dlss/was/TestWasapiDownloader.java

```diff
@@ -178,4 +179,51 @@ public void downloadSelectedWarcs_byJobIdLowerBound() throws Exception {
     verify(mockCrawlSelector).getSelectedCrawlIds(Integer.valueOf(argValue));
   }
 
+  @Test
+  public void prepareOutputLocation_correctLocation() throws SettingsLoadException {
+    WasapiDownloader wd = new WasapiDownloader(WasapiDownloader.SETTINGS_FILE_LOCATION, null);
+    WasapiFile wfile = new WasapiFile();
+    String collId = "123";
+    wfile.setCollectionId(Integer.parseInt(collId));
+    String crawlStartTime = "2017-01-01T00:00:00Z";
+    wfile.setCrawlStartDateStr(crawlStartTime);
+    String filename = "i_is_a_warc_file";
+    wfile.setFilename(filename);
+    String crawlId = "666";
+    wfile.setCrawlId(Integer.parseInt(crawlId));
+    WasapiDownloaderSettings mySettings = new WasapiDownloaderSettings(WasapiDownloader.SETTINGS_FILE_LOCATION, null);
+
+    String result = wd.prepareOutputLocation(wfile);
+    String expected = mySettings.outputBaseDir() + "AIT_" + collId + "/" + crawlId + "/" + crawlStartTime + "/" + filename;
+    assertEquals("Incorrect output location", expected, result);
+  }
+
+  @Test
+  public void prepareOutputLocation_whenMissingJsonValues() throws SettingsLoadException {
+    WasapiDownloader wd = new WasapiDownloader(WasapiDownloader.SETTINGS_FILE_LOCATION, null);
+    String result = wd.prepareOutputLocation(new WasapiFile());
+    assertThat(result, org.hamcrest.CoreMatchers.containsString("/AIT_0/")); // when missing collectionId
+    assertThat(result, org.hamcrest.CoreMatchers.containsString("/0/")); // when missing crawlId
+    assertThat(result, org.hamcrest.CoreMatchers.containsString("/null/")); // when missing crawlStartTime
+    assertThat(result, org.hamcrest.CoreMatchers.endsWith("/null")); // when missing filename - gah!
```

Comment:
fine by me.  one thing i wasn't considering when i was looking at it, is that it's unlikely that these will all have default values (right?), so at least there's some separation due to the other properties.